flux-top: honor FLUX_F58_FORCE_ASCII

chu11 opened the pull request:
Problem: flux-top does not honor the FLUX_F58_FORCE_ASCII environment variable and outputs ƒ ("fancy-f") all the time.

Output "f" instead of ƒ when FLUX_F58_FORCE_ASCII is set.

Fixes https://github.com/flux-framework/flux-core/issues/4837

Note, I added one dumb test for coverage, borderline maybe isn't even necessary.  Ultimately grepping for ƒ from the `flux-top` was hard given whatever format curses + runpty ultimately put things in.  Grepping for normal-f seemed pointless given there's going to be a bunch of `f` anyways via normal hex output.

grondo reviewed (approved):
Looks good to me. One potential simplification inline. but take it or leave it as far as I'm concerned.

chu11 commented on src/cmd/top/summary_pane.c:
```diff
@@ -95,7 +95,10 @@ static void draw_timeleft (struct summary_pane *sum)
 {
     wattron (sum->win, COLOR_PAIR (TOP_COLOR_YELLOW));
-    mvwprintw (sum->win, level_dim.y_begin, level_dim.x_begin, "ƒ");
+    if (sum->top->f_ascii)
+        mvwprintw (sum->win, level_dim.y_begin, level_dim.x_begin, "f");
+    else
+        mvwprintw (sum->win, level_dim.y_begin, level_dim.x_begin, "ƒ");
```
oh good idea

chu11 commented:
re-pushed, going with the suggested approach from @grondo above

```C
+    if (getenv ("FLUX_F58_FORCE_ASCII"))
+        top->f_char = "f";
+    else
+        top->f_char = "ƒ";
```

chu11 commented:
re-pushed, and also added a fix to ensure the `f_char` is recursively carried down into other jobs selected.  tested by hand to ensure was issue in prior commits and is now fixed.

@grondo perhaps want to take a skim and LMK what you think.  Basically just pass in potential parent to `top_create()` and initialize `f_char` from it if it was passed in.

grondo commented:
>  Basically just pass in potential parent to top_create() and initialize f_char from it if it was passed in.

Shouldn't `top->f_char` be initialized in `top_create()`? The environment isn't going to change so it would "just work" without needing to pass in the parent top object?

chu11 commented:
> Shouldn't top->f_char be initialized in top_create()? The environment isn't going to change so it would "just work" without needing to pass in the parent top object?

True, I guess I was thinking ahead to other potential "absorb from parent" uses.  But no need to jump ahead too far.

grondo commented:
> True, I guess I was thinking ahead to other potential "absorb from parent" uses. But no need to jump ahead too far.

That's a good point, but you're right it could be added when we start having other top options that need to be inherited by recursive top.

chu11 commented:
re-pushed with a much simpler solution

chu11 commented:
rebased & re-pushed, setting MWP
